AsyncDisplayKit/ASCollectionView.mm

```diff
@@ -906,25 +930,32 @@ - (UICollectionViewCell *)collectionView:(UICollectionView *)collectionView cell
   UICollectionViewCell *cell = nil;
   ASCellNode *node = [self nodeForItemAtIndexPath:indexPath];
 
-  if (node.shouldUseUIKitCell && _asyncDataSourceFlags.interop) {
+  BOOL shouldDequeueExternally = _asyncDataSourceFlags.interopAlwaysDequeue || (_asyncDataSourceFlags.interop && node.shouldUseUIKitCell);
+  if (shouldDequeueExternally) {
     cell = [(id<ASCollectionDataSourceInterop>)_asyncDataSource collectionView:collectionView cellForItemAtIndexPath:indexPath];
   } else {
-    ASDisplayNodeAssert(node != nil, @"Cell node should exist. indexPath = %@, collectionDataSource = %@", indexPath, self);
     cell = [self dequeueReusableCellWithReuseIdentifier:kReuseIdentifier forIndexPath:indexPath];
-    [(_ASCollectionViewCell *)cell setNode:node];
+  }
+
+  ASDisplayNodeAssert(node != nil, @"Cell node should exist. indexPath = %@, collectionDataSource = %@", indexPath, self);
+
+  if (_ASCollectionViewCell *asCell = ASDynamicCast(cell, _ASCollectionViewCell)) {
```

Comment:
Nice usage of ASDynamicCast!  Slightly less of a fan of doing the assignment inside the (), but I'm guessing you love it :) (I'm not opposed to it landing)